test/adapter_test.rb

```diff
@@ -0,0 +1,58 @@
+require 'test_helper'
+
+class TestSemianAdapter < Minitest::Test
+  def setup
+    Semian.unregister_all_resources
+    # Consumers registered in other test files must be cleared
+    Semian.instance_variable_set(:@consumers, nil)
+    GC.enable
+    GC.start
+  end
+
+  def teardown
+    Semian.unregister_all_resources
+  end
+
+  def test_adapter_registers_consumer
+    assert_equal(true, Semian.resources.empty?)
+    assert_equal(true, Semian.consumers.empty?)
+    client = Semian::AdapterTestClient.new(quota: 0.5)
+    resource = client.semian_resource
+    assert_equal(resource, Semian.resources[client.semian_identifier])
+    assert_equal(client.inspect, Semian.consumers[client.semian_identifier].first.inspect)
+  end
+
+  def test_unregister_all_resources
+    client = Semian::AdapterTestClient.new(quota: 0.5)
+    resource = client.semian_resource
+    assert_equal(resource, Semian.resources[client.semian_identifier])
+    assert_equal(client.inspect, Semian.consumers[client.semian_identifier].first.inspect)
+
+    GC.disable # need to disable GC to ensure client weak reference is alive for assertion below
+    Semian.unregister_all_resources
+
+    assert_equal(true, Semian.resources.empty?)
+    assert_equal(true, Semian.consumers.empty?)
+    assert_nil client.instance_variable_get(:@semian_resource)
```

Comment:
@fw42 I agree, but I'm not sure the best way to work around this.

Calling semian_resource will create a new one, which affects what I'm trying to test here.

Any ideas?